pkg/processor/metricfrequencyprocessor/README.md

```diff
@@ -0,0 +1,40 @@
+## <a name="metricfrequencyprocessor"></a>Metric Frequency Processor
+
+The `metricfrequencyprocessor` is a metrics processor that helps reduce DPM by automatic tuning of metrics reporting frequency which adjusts for metric's information volume. 
+
+For `metricfrequencyprocessor`, there are three categories of metrics - constant, low information and regular.
+Constant ones are self-explanatory. 
+Low information ones are defined by heuristic - having no anomalous data points and having small relative variation.
+All metrics not falling to above categories are regular and do not have their report frequency tuned beyond `maxReportFrequency`.
+
+Metrics are categorised by their recent data points, so a category for a metric can can change.
+
+`metricfrequencyprocessor` works by sifting out data points that would be reported earlier than according to their category's frequency.  
+
+### Config
+
+- `min_point_accumulation_seconds` - warm up time for processor. Processor won't sift any data point from a metric with no earlier data point older than this value.     
+- `constant_metrics_report_frequency_seconds` - minimum time between reports of a constant metric.
+- `low_info_metrics_report_frequency_seconds` - minimum time between reports of a low info metric.
+- `max_report_frequency_seconds` - minimum time between reports of any metric.
+  
+*Low info definition*
+- `iqr_anomaly_coefficient` - relative deviation from interquartile range which constitutes an anomaly.
+- `variation_iqr_threshold_coefficient` - variation to iqr quotient under which a metric is considered low info. 
+  
+*Data point caching*
+- `data_point_expiration_minutes` - how long a data point should be used for determining metrics category. 
+- `data_point_cache_cleanup_interval_minutes` - how often expired data points are removed from memory.
+- `metric_cache_cleanup_interval_minutes` - how often no longer seen metrics are removed from memory.
+
+### Example config
+
+```yaml
+processors:
+  metric_frequency:
+    min_point_accumulation_seconds: 900
```

Comment:
time.Duration can be used as type, it will implicate that e.g. `60s` can be used as value